src/components/Search/SearchComponent.tsx

```diff
@@ -87,33 +71,29 @@ function SearchComponent({
       },
     },
   });
-  const cancelBtn = {
-    main: style({
-      $nest: {
-        '&.MuiButtonBase-root': {
-          ...btnStyles.Secondary,
-        },
-      },
-    }),
-  };
 
-  const dispatch = useAppDispatch();
+  const { search } = useAppSelector((state) => state);
```

Comment:
Selectors shouldn't return the full state, because this makes your component rerender each time any part of the state changes. Instead you should return the smallest part of the state you need, and even do the computation that returns exactly what you need inside the selector. It's OK to have several `useAppSelector` calls.